File: AFSwinx/src/main/java/com/tomscz/afswinx/rest/connection/HeaderType.java
package com.tomscz.afswinx.rest.connection;

/**
 * This class specify type of supported request and response types.
 * 
 * @author Martin Tomasek (martin@toms-cz.com)
 * 
 * @since 1.0.0.
 */
public enum HeaderType {
    // XML is not supported in this version, but it will be supported soon.
    XML("application/xml"), JSON("application/json");

    private final String name;

    private HeaderType(String name) {
        this.name = name;
    }

    public boolean equalsName(String otherName) {
        return (otherName == null) ? false : name.equals(otherName);
    }

    public String toString() {
        return name;
    }

}

File: AFMobile/AFWinPhone/AFWinPhone/rest/connection/HeaderType.cs
using System;

namespace AFWinPhone.rest.connection
{
    public sealed class HeaderType
    {

        public static readonly HeaderType XML = new HeaderType("application/xml");
        public static readonly HeaderType JSON = new HeaderType("application/json");

        private String name;

        private HeaderType(String name)
        {
            this.name = name;
        }

        public override String ToString()
        {
            return name;
        }
    }
}
